pkg/graph/resolvers/plan_data_exchange_approach.go

```diff
@@ -62,7 +62,28 @@ func PlanDataExchangeApproachUpdate(
 		return nil, err
 	}
 
+	targetStatus, err := EvaluateStatus(existing)
+	if err != nil {
+		return nil, err
+	}
+
+	existing.Status = &targetStatus
+
 	// Update the plan data exchange approach
 	retDataExchangeApproach, err := store.PlanDataExchangeApproachUpdate(logger, existing)
 	return retDataExchangeApproach, err
 }
+
+// EvaluateStatus derives the status of the data exchange approach based on the current state of the model
+func EvaluateStatus(dea *models.PlanDataExchangeApproach) (models.DataExchangeApproachStatus, error) {
+
+	if dea.ModifiedDts == nil {
+		return models.DataExchangeApproachStatusNotStarted, nil
+	}
```

Comment:
I don't think `ModifiedDts` is ever set as part of the Update function prior to this function being called.

I'd probably recommend that instead of calculating/evaluating the status based on the `models.PlanDataExchangeApproach`, we could probably just add a clause in the resolver itself that uses `isSettingToComplete` to determine which of the 2 statuses to set (`COMPLETE` if `isSettingToComplete`, otherwise `IN_PROGRESS`)

Reply:
🤔 I see your point. Ok, 2 minutes.